### PR TITLE
bump solang-parser to 0.3.1

### PR DESCRIPTION
**Pull request by ArielTM**
@mattsse please check this with a grain of salt. I'm not sure what the problems were, but it compiles and all tests passed except for the following:
```
    cmd::can_check_snapshot
    create::can_create_template_contract
    create::can_create_using_unlocked
    test_cmd::can_use_libs_in_multi_fork
```
But failure seems unrelated (number of contracts compiled)

## Motivation

(Hopefully) Closes https://github.com/foundry-rs/foundry/issues/5358

## Solution

Upgrade

### CODE REVIEW

File: chisel/src/solidity_helper.rs [hunk 1]
```diff
@@ -158,13 +157,7 @@ impl SolidityHelper {
...
         for res in Lexer::new(input, 0, &mut comments, &mut errors) {
             match res {
-                Err(err) => match err {
-                    LexicalError::EndOfFileInComment(_) |
-                    LexicalError::EndofFileInHex(_) |
-                    LexicalError::EndOfFileInString(_) => return ValidationResult::Incomplete,
-                    _ => return ValidationResult::Valid(None),
-                },
-                Ok((_, token, _)) => match token {
+                (_, token, _) => match token {
```
Comment (DaniPopes) [inline chisel/src/solidity_helper.rs:160]: can replace `res` in the `for` above with this pattern and remove the match entirely

```
for (_, token, _) in ...
```

**Review by DaniPopes — APPROVED** [review]
thx
cc @mattsse @Evalir

**Review by Evalir — APPROVED** [review]
lgtm! thanks!

**Review by mattsse — APPROVED** [review]
lgtm,

unrelated clippy

defer to @Evalir wen merge

**Pull request merged**